search-replace.go

```diff
@@ -155,71 +162,229 @@ func fixLine(line *[]byte, replacements []*Replacement) *[]byte {
+			return nil, fmt.Errorf("faulty data, byte count does not match data size")
+		}
+		char := linePart[currentContentIndex]
+		secondChar := linePart[currentContentIndex+1]
+		thirdChar := linePart[currentContentIndex+2]
+		if char == backslash && contentByteCount < details.OriginalByteSize {
+			unescapedBytePair := getUnescapedBytesIfEscaped(linePart[currentContentIndex : currentContentIndex+2])
+			// if we get the byte pair without the backslash, it corresponds to a byte
+			contentByteCount += len(unescapedBytePair)
+
+			// content index count remains the same.
+			currentContentIndex += 2
 			continue
 		}
 
-		// Calculate the new length and update the serialized length prefix
-		newLength := len(updatedContent)
-		newLengthStr := []byte(strconv.Itoa(newLength))
-		Debugf("Replaced content new length: %d\n", newLength)
-		*line = append((*line)[:startIndex+match[2]], append(newLengthStr, (*line)[startIndex+match[3]:]...)...)
-		Debugf("After updating length prefix: %s\n", string(*line))
+		if char == backslash && secondChar == quote && thirdChar == semicolon && contentByteCount >= details.OriginalByteSize {
 
-		// Update the serialized content inline
+			// since we've filtered out all the escaped value already, this should be the actual end
+			nextPartIndex = currentContentIndex + 3
+			details.NextPartIndex = nextPartIndex
+			// we're at backslash, so we need to minus 1 to get the index where the content finishes
+			details.ContentEndIndex = currentContentIndex - 1
+			nextPartFound = true
+			break
+		}
+
+		contentByteCount++
+		currentContentIndex++
+	}
+
+	if nextPartFound == false {
+		return nil, fmt.Errorf("end of serialized string not found")
+	}
+
+	return &details, nil
+}
 
-		//contentEnd = contentStart + newLength // adjust the end index based on the new length -- THIS BREAKS THINGS
+func getUnescapedBytesIfEscaped(charPair []byte) []byte {
 
-		*line = append((*line)[:contentStart], append(updatedContent, (*line)[contentEnd:]...)...)
-		Debugf("After updating content: %s\n", string(*line))
+	backslash := byte('\\')
 
-		// Adjust startIndex for the next iteration
-		startIndex += match[1] + newLength + len(newLengthStr) - len((*line)[startIndex+match[2]:startIndex+match[3]])
-		Debugf("New startIndex: %d\n", startIndex)
+	//escapables := []byte{'\\', '\'', '"', 'n', 'r', 't', 'b', 'f', '0'}
+
+	// a map of the second byte to its actual binary presentation
+
+	// if the first byte is not a backslash, we don't need to do anything
+
+	if charPair[0] != backslash {
+		return charPair
 	}
 
-	return line
+	unescapedMap := map[byte]byte{
+		'\\': '\\',
+		'\'': '\'',
+		'"':  '"',
+		'n':  '\n',
+		'r':  '\r',
+		't':  '\t',
+		'b':  '\b',
+		'f':  '\f',
+		'0':  '\x00',
+	}
+
+	actualByte := unescapedMap[charPair[1]]
+
+	if actualByte != 0 {
+		return []byte{actualByte}
+	}
+
+	// what if it's not a valid escape? Do nothing - it's considered as already escaped
+	return charPair
```

Comment:
I need to double check if this is correct - admittedly I had GH Copilot fill this up.

Reply:
Verified, should be correct.